### PR TITLE
Adds example how to force model hook to execute

### PR DESCRIPTION
**Pull request by locks**
I think this would be useful, since it's a recurring question I see from embereños.
Should I elaborate it a bit more and have a separate code sample, or is something like this enough?

### CODE REVIEW

**Comment by jfelchner** [comment]
:+1: 

For people new to Ember an explicit `link-to` might also be helpful.

```
{{link-to 'my_route' my_object}}  // Does not force the model hook to fire
{{link-to 'my_route' my_object.id}}  // Forces the model hook to fire
```

**Comment by locks** [comment]
@rwjblue not sure why the build failed, but can you merge?

**Comment by stefanpenner** [comment]
kicking tests now.

File: packages/ember-routing/lib/system/route.js [hunk 1]
```diff
@@ -1302,6 +1302,11 @@ var Route = EmberObject.extend(ActionHandler, Evented, {
 
     // integer passed in, model hook is called
     this.transitionTo('post', 1);
+
```
Comment (rwjblue) [inline packages/ember-routing/lib/system/route.js]: Tests are failing for trailing whitespace here (and possibly below).

**Comment by locks** [comment]
How can I squash the commits? I can't figure it out since it's a `patch-1` sort of deal.

**Comment by stefanpenner** [comment]
you'll have to fetch it locally and 

i have ` pr = "!f() { git fetch origin refs/pull/$1/head:pr-$1 && git checkout pr-$1; } ; f"` in my .gitconfig under `[alias]`

**Comment by locks** [comment]
According to https://github.com/emberjs/ember.js/commit/f0713b32860f18dc6753e96437753dc531869fc4 I think we can ignore [the failing test](https://travis-ci.org/emberjs/ember.js/jobs/70473038#L210) and merge this.

**Comment by mixonic** [comment]
kicked it. I'd rather not merge it if the build is not green, makes it harder to trust any given PR

**Pull request merged**